src/OrchardCore.Modules/OrchardCore.AuditTrail/Controllers/ContentController.cs

```diff
@@ -0,0 +1,135 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Localization;
+using OrchardCore.Admin;
+using OrchardCore.AuditTrail.Extensions;
+using OrchardCore.AuditTrail.Indexes;
+using OrchardCore.AuditTrail.Models;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Display;
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.Contents;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.DisplayManagement.Notify;
+using YesSql;
+using IYesSqlSession = YesSql.ISession;
+
+namespace OrchardCore.AuditTrail.Controllers
+{
+    [Admin]
+    public class ContentController : Controller
+    {
+        private readonly IHtmlLocalizer H;
+        private readonly INotifier _notifier;
+        private readonly IYesSqlSession _session;
+        private readonly IHttpContextAccessor _hca;
+        private readonly IContentManager _contentManager;
+        private readonly IUpdateModelAccessor _updateModelAccessor;
+        private readonly IAuthorizationService _authorizationService;
+        private readonly IContentItemDisplayManager _contentItemDisplayManager;
+
+
+        public ContentController(
+            IYesSqlSession session,
+            INotifier notifier,
+            IHttpContextAccessor hca,
+            IContentManager contentManager,
+            IUpdateModelAccessor updateModelAccessor,
+            IAuthorizationService authorizationService,
+            IHtmlLocalizer<ContentController> htmlLocalizer,
+            IContentItemDisplayManager contentItemDisplayManager)
+        {
+            _hca = hca;
+            _session = session;
+            _notifier = notifier;
+            _contentManager = contentManager;
+            _updateModelAccessor = updateModelAccessor;
+            _authorizationService = authorizationService;
+            _contentItemDisplayManager = contentItemDisplayManager;
+
+            H = htmlLocalizer;
+        }
+
+
+        public async Task<ActionResult> Detail(int versionNumber, string auditTrailEventId)
+        {
+            var auditTrailEvent = await _session.Query<AuditTrailEvent, AuditTrailEventIndex>()
+                .Where(auditTrailEventIndex => auditTrailEventIndex.AuditTrailEventId == auditTrailEventId)
+                .FirstOrDefaultAsync();
+
+            var contentItem = auditTrailEvent.Get(auditTrailEvent.EventName).ToObject<ContentItem>();
+
+            if (!await _authorizationService.AuthorizeAsync(User, CommonPermissions.ViewContent, contentItem))
```

Comment:
Is `ViewContent` permission appropriate here? I know it will be read-only but it will be a fully rendered editor shape with read-only fields.